src/util/hash_util.rs

```diff
@@ -17,11 +17,18 @@
 
 use data_type::AsBytes;
 
+#[cfg(target_feature="sse4.2")]
+use x86intrin::sse42;
+
 /// Computes hash value for `data`, with a seed value `seed`.
 /// The data type `T` must implement the `AsBytes` trait.
-/// TODO: implement more efficient hash, such as Crc32, using SSE4 instructions.
-pub fn hash<T: AsBytes>(data: &T, seed: u64) -> u64 {
-  murmur_hash2_64a(data, seed)
+pub fn hash<T: AsBytes>(data: &T, seed: u32) -> u32 {
+  #[cfg(target_feature="sse4.2")] {
+    crc32_hash(data, seed)
+  }
+  #[cfg(not(target_feature="sse4.2"))] {
```

Comment:
Have you considered making another flag as `target_feature="murmur2"` and also have a default value as "murmur2"? IMHO, this would allow to select murmur2 explicitly, rather achieving it by not selecting sse4.2. Just a thought, it looks good either way.

Reply:
I think `target_features` are hardware features and they are fixed (see [here](https://github.com/rust-lang/rust/issues/29717) for a complete list). The purpose here is that if the target machine support SSE4.2, then during compilation we can enable the faster CRC32C hashing, otherwise we use Murmur hashing. Since there're only two choices, it is not that necessary to define another feature.